cpp/src/groupby/sort/scan.cpp

```diff
@@ -144,6 +144,23 @@ void scan_result_functor::operator()<aggregation::DENSE_RANK>(aggregation const&
     detail::dense_rank_scan(
       order_by, helper.group_labels(stream), helper.group_offsets(stream), stream, mr));
 }
+
+template <>
+void scan_result_functor::operator()<aggregation::PERCENT_RANK>(aggregation const& agg)
+{
+  if (cache.has_result(values, agg)) return;
+  CUDF_EXPECTS(helper.is_presorted(),
+               "Percent rank aggregate in groupby scan requires the keys to be presorted");
```

Comment:
Why? Can we just do the sorting now?

Reply:
No, we don't want to add sorting here. We do not want to alter the row order at this point, for any of the ranking functions.

For the Spark use case that this addresses, the data would already have been sorted upstream. We might be receiving a batch here from a larger group, sorted by columns that aren't even available here.

Reply:
Fixed.